code_generation/enum.cpp

```diff
@@ -88,7 +88,18 @@ QString Enum::declaration() const
         }
     }
 
+    if (!d->mEnums.isEmpty())
+        retval += QStringLiteral(", ");
+    if (d->mCombinable)
+        retval += QString("Invalid = %1").arg(value);
+    else
+        retval += QStringLiteral("Invalid");
```

Comment:
What if the enum already contains a value called Invalid?
More generally, what if the programmer does NOT want an Invalid value to be added?

I think this should be up to the user of the Enum class to add an Invalid value if it wants to.

Reply:
That's both true, I will move it upwards as well. In the kxml_compiler generated codes I use this value to initialise the generated code's variables.